src/app/api/products/route.ts

```diff
@@ -0,0 +1,27 @@
+import { prisma } from "@/utils/connect";
+import { NextRequest, NextResponse } from "next/server";
+
+//fetch all of the products 
+export const GET = async (req:NextRequest) => {
+
+    const {searchParams} = new URL(req.url)
+    const cat = searchParams.get("cat")
+    try {
+
+        const products = await prisma.product.findMany({
+            where:{
+                ...(cat ? {catSlug:cat} : {isFeatured:true}),
+            }
+        });
+
+        return new NextResponse(JSON.stringify(products), {status:200});
+
+    } catch (error) {
+        console.log(error);
+        return new NextResponse(JSON.stringify({message:"Something went wrong!"}), {status:500});
```

Comment:
You can create a separate enum for storing error messages, the other one for statutes. It'd be useful, I think

Reply:
Oh okay will do